src/main/java/com/urbanairship/datacube/idservices/MapIdService.java

```diff
@@ -75,7 +75,7 @@
     }
 
     @Override
-    public Optional<byte[]> getId(int dimensionNum, byte[] bytes, int numIdBytes) throws IOException, InterruptedException {
+    synchronized public Optional<byte[]> getId(int dimensionNum, byte[] bytes, int numIdBytes) throws IOException, InterruptedException {
```

Comment:
probably paranoia on my part, but it seemed like we were seeing some odd results from a map harness test.